rust/lance/src/io/reader.rs

```diff
@@ -1528,22 +1529,22 @@ mod tests {
 
         let arr = read_array_w_params(&reader, &schema.fields[1], ReadBatchParams::RangeFull).await;
         assert_eq!(100, arr.len());
-        assert_eq!(100, arr.null_count());
+        assert_eq!(arr.data_type(), &DataType::Null);
```

Comment:
`null_count()` no longer returns the logical null count, just the number of "physical nulls". https://github.com/apache/arrow-rs/pull/4691